### DISCUSSION TITLE
Problem with Building

### DISCUSSION THREAD

**Discussion by DuckyBlender** [NVlabs/instant-ngp#296] [Q&A]
Log file: https://pastebin.com/uZCE9uE0
Won't compile successfully.
Using Windows 10 and Python 3.9.10
Just downloaded lastest CMake, CUDA and OptiX
GTX 1660 SUPER

**Comment by bolshoytoster** [comment]
@DuckyBlender does [this](https://github.com/NVlabs/tiny-cuda-nn/issues/10#issuecomment-1012778211) help?

**Comment by quinwu** [comment]
I had the same problem, my GPU is 3090

System windows10
Python 3.9.10
CUDA 11.6.55
optiX 7.4.0

**Comment by bolshoytoster** [comment]
@quinwu does [this](https://github.com/NVlabs/instant-ngp/issues/213#issuecomment-1047708565) help?

**Reply by pratikpadamwar** [reply]
I have set the TCNN_CUDA_ARCHITECTURES=86 so i was able to build with cmake but after build 
when I run script/run.py am getting this following error 
 ERROR  : Uncaught exception**: FullyFusedMLP: insufficient shared memory available on the GPU. Reduce `n_neurons` or use `CutlassMLP` (better compatibility but slower)** 
is this error related to pytorch shared memory problem 
I tried everything but it is not working 
my GPU size 4096 MB 
can anyone please help me out here

**Reply by Tom94** [reply]
You need to go into the `.json` config you're using (likely `nerf/base.json` if you're loading a NeRF dataset and not specifying a config) and replace `FullyFusedMLP` with `CutlassMLP`.

As the error message says, the GPU you have installed likely has too little shared memory for `FullyFusedMLP`. More info on shared memory here: https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#memory-hierarchy

**Comment by quinwu** [comment]
> @quinwu does [this](https://github.com/NVlabs/instant-ngp/issues/213#issuecomment-1047708565) help?

export TCNN_CUDA_ARCHITECTURES=86 is not a Windows command, I don't know how to change it

**Comment by bolshoytoster** [comment]
@quinwu it's an environment variable, how are you running the compiler?

**Comment by quinwu** [comment]
```
-- Performing Test HAS_MSVC_GL_LTCG - Success
-- Obtained target architecture from environment variable TCNN_CUDA_ARCHITECTURES=86
-- Targeting GPU architectures: 86
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/yckj/Desktop/instant-ngp/build
```
I set CNN_CUDA_ARCHITECTURES=86 as environment variable, still have the same problem.

> @DuckyBlender does [this](https://github.com/NVlabs/tiny-cuda-nn/issues/10#issuecomment-1012778211) help?

**Comment by bolshoytoster** [comment]
@quinwu can you paste the ful error please.

**Comment by quinwu** [comment]
> @quinwu can you paste the ful error please.

https://pastebin.com/hRVYhKZW

**Comment by bolshoytoster** [comment]
@quinwu did you delete the `build/` directory and start over?

**Comment by quinwu** [comment]
> @quinwu did you delete the `build/` directory and start over?

yes, delete build/ and start over

**Comment by DuckyBlender** [comment]
> @DuckyBlender does [this](https://github.com/NVlabs/tiny-cuda-nn/issues/10#issuecomment-1012778211) help?

Still doesn't work
https://pastebin.com/7X3x9QZf
![SystemPropertiesAdvanced_RsKdSuAXGA](https://user-images.githubusercontent.com/42645784/155434259-d33d07e4-683f-4252-8ef5-9960fa30e066.png)

I made a work around by getting a compiled build from my friend.

**Comment by quinwu** [comment]
> 



> > @DuckyBlender does [this](https://github.com/NVlabs/tiny-cuda-nn/issues/10#issuecomment-1012778211) help?
> 
> Still doesn't work https://pastebin.com/7X3x9QZf ![SystemPropertiesAdvanced_RsKdSuAXGA](https://user-images.githubusercontent.com/42645784/155434259-d33d07e4-683f-4252-8ef5-9960fa30e066.png)
> 
> I made a work around by getting a compiled build from my friend.


compiled build tcnn?

**Comment by DuckyBlender** [comment]
> > 
> 
> > > @DuckyBlender does [this](https://github.com/NVlabs/tiny-cuda-nn/issues/10#issuecomment-1012778211) help?
> > 
> > 
> > Still doesn't work https://pastebin.com/7X3x9QZf ![SystemPropertiesAdvanced_RsKdSuAXGA](https://user-images.githubusercontent.com/42645784/155434259-d33d07e4-683f-4252-8ef5-9960fa30e066.png)
> > I made a work around by getting a compiled build from my friend.
> 
> compiled build tcnn?

what does tcnn mean?

**Comment by quinwu** [comment]
> > > 
> > 
> > 
> > > > @DuckyBlender does [this](https://github.com/NVlabs/tiny-cuda-nn/issues/10#issuecomment-1012778211) help?
> > > 
> > > 
> > > Still doesn't work https://pastebin.com/7X3x9QZf ![SystemPropertiesAdvanced_RsKdSuAXGA](https://user-images.githubusercontent.com/42645784/155434259-d33d07e4-683f-4252-8ef5-9960fa30e066.png)
> > > I made a work around by getting a compiled build from my friend.
> > 
> > 
> > compiled build tcnn?
> 
> what does tcnn mean?

tiny-cuda-nn

**Reply by nowhereman1999** [reply]
Hi I seems to meet the same problem like this:
![image](https://user-images.githubusercontent.com/53631206/171829893-be3fdf47-e31c-4364-89ef-dccf0b257f20.png)
and I have searched the error in the issues and found [https://github.com/NVlabs/instant-ngp/issues/178](https://github.com/NVlabs/instant-ngp/discussions/url) which says it can be worked out by install the cuda 11.6 ,However,the current cuda installed on my computer is 11.3 and I checked highest cuda version supported by my RTX2060 is 11.4 which is shown in the figure below:
![image](https://user-images.githubusercontent.com/53631206/171830030-49b229f7-b3be-40d9-a793-762cd8317e6f.png)
So I even can't try to install a new cuda 11.6 to run the command again and find out wether it works,have you meet the same problem with me and could you please give me some advice?Looking forward to your reply:-)

**Reply by bolshoytoster** [reply]
@Hezhexi2002 the screenshot doesn't actually show any error, it just says that there were errors.

Could you paste the full error here (in triple backticks:
\`\`\`
paste here
\`\`\`

**Reply by nowhereman1999** [reply]
OK,here is my CMakeOutput.log and CMakeError.log:
[CMakeOutput.log](https://github.com/NVlabs/instant-ngp/files/8831636/CMakeOutput.log)
[CMakeError.log](https://github.com/NVlabs/instant-ngp/files/8831638/CMakeError.log)

**Reply by bolshoytoster** [reply]
@Hezhexi2002 does [this](https://docs.microsoft.com/en-us/answers/questions/76739/fatal-error-c1083-cannot-open-include-file-39pthre.html) help?

**Reply by nowhereman1999** [reply]
Sorry,the logs I have presented before seems not the true reason of the error,I try to generate the project in the VS again and  increase verbosity of VS output,here is the output log:
[ALL_BUILD.log](https://github.com/NVlabs/instant-ngp/files/8831735/ALL_BUILD.log)

**Reply by nowhereman1999** [reply]
I‘m so glad to tell you that I have successfully compiled the project after I install the CUDAv11.6,it seems that this is really due to the conflict between the nvcc of CUDAv11.3 and the MSVC of VS2019:
![image](https://user-images.githubusercontent.com/53631206/172034658-1f2feeea-b438-4621-bf33-8d526b33f7e5.png)

**Comment by DuckyBlender** [comment]
> > > > 
> 
> > > 
> 
> > > 
> 
> > > > > @DuckyBlender does [this](https://github.com/NVlabs/tiny-cuda-nn/issues/10#issuecomment-1012778211) help?
> 
> > > > 
> 
> > > > 
> 
> > > > Still doesn't work https://pastebin.com/7X3x9QZf ![SystemPropertiesAdvanced_RsKdSuAXGA](https://user-images.githubusercontent.com/42645784/155434259-d33d07e4-683f-4252-8ef5-9960fa30e066.png)
> 
> > > > I made a work around by getting a compiled build from my friend.
> 
> > > 
> 
> > > 
> 
> > > compiled build tcnn?
> 
> > 
> 
> > what does tcnn mean?
> 
> 
> 
> tiny-cuda-nn

Idk man i just got the build folder from my friend and replaced mine.
